### PR TITLE
Update doc comments to conform to eslint-complete-sentences rule

### PR DESCRIPTION
**Pull request by ryanhamley**
## Launch Checklist

Documenting two pain points here:

1) The line that is referenced in the lint error may not exactly correspond to the line in your editor where the error actually is. This is because eslint reads all consecutive comment lines as one block that is denoted in a potential error message by the line the block starts on. This allows for multi-line comments, but it tripped me up at first looking for the issue.

2) eslint will consider _any_ period to be the end of a sentence. This means that this rule will error on something like `e.g.`, `i.e.` or `object.method()`. In general, I think we should avoid e.g. and i.e. anyway and we could always use `object#method` or just `method` in most comments.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -199,7 +199,7 @@ const exported = {
      * Takes precedence over `mapboxgl.workerUrl`.
      *
      * @var {Object} workerClass
-     * @returns {Object|null} a Class object, an instance of which exposes the `Worker` interface.
+     * @returns {Object|null} Class object, an instance of which exposes the `Worker` interface.
```
Comment (ansis) [inline src/index.js:202]:
```suggestion
     * @returns {Object|null} A class which implements the `Worker` interface.
```

File: src/source/geojson_source.js [hunk 2]
```diff
@@ -192,10 +192,10 @@ class GeoJSONSource extends Evented implements Source {
      * For clustered sources, fetches the original points that belong to the cluster (as an array of GeoJSON features).
      *
      * @param clusterId The value of the cluster's `cluster_id` property.
-     * @param limit The maximum number of features to return. (Defaults to `10` if a falsy value is given.)
-     * @param offset The number of features to skip (e.g. for pagination). (Defaults to `0` if a falsy value is given.)
+     * @param limit The maximum number of features to return. (Defaults to `10` if a falsy value is given).
+     * @param offset The number of features to skip (for use with pagination, for example). (Defaults to `0` if a falsy value is given).
```
Comment (ansis) [inline src/source/geojson_source.js:196]:
```suggestion
     * @param offset The number of features to skip. This is useful for paginating results. (Defaults to `0` if a falsy value is given).
```

**Review by ansis — COMMENTED** [review]
Overall, looks great. I added a couple suggestions. One of them as a commit (518124533b13c9d109656a84a7a726e968c63220) but feel free to revert that if you disagree.

The only larger thing to check is which changes in .eslintrc are intended for this pr

File: .eslintrc [hunk 3]
```diff
@@ -22,10 +22,10 @@
     // no-duplicate-imports doesn't play well with Flow
     // https://github.com/babel/eslint-plugin-babel/issues/59
     "no-duplicate-imports": "off",
-    "import/no-duplicates": "error",
+    "import/no-duplicates": "off",
```
[the remark this answers is not in the document]
Comment (ryanhamley) [reply .eslintrc:25]: Yes, I thought we were supposed to push like this and then revert after approval

File: src/index.js (hunk 1 above)
Comment (ryanhamley) [reply src/index.js]: Fixed in 7251155

File: src/source/geojson_source.js (hunk 2 above)
Comment (ryanhamley) [reply src/source/geojson_source.js]: Fixed in 7251155

File: .eslintrc (hunk 3 above)
[the remark this answers is not in the document]
Comment (ryanhamley) [reply .eslintrc]: Reverted in 641a8d7

File: src/index.js [hunk 4]
```diff
@@ -240,7 +240,7 @@ Debug.extend(exported, {isSafari, getPerformanceMetrics: PerformanceUtils.getPer
  * @param {Object} [options]
  * @param {boolean} [options.failIfMajorPerformanceCaveat=false] If `true`,
  *   the function will return `false` if the performance of Mapbox GL JS would
- *   be dramatically worse than expected (e.g. a software WebGL renderer would be used).
+ *   be dramatically worse than expected (a software WebGL renderer would be used).
```
Comment (karimnaaji) [inline src/index.js:243]: nit: remove extra space
```suggestion
 *  the function will return `false` if the performance of Mapbox GL JS would
 *  be dramatically worse than expected (a software WebGL renderer would be used).
```

File: src/source/tile_cache.js [hunk 5]
```diff
@@ -27,9 +27,9 @@ class TileCache {
     }
 
     /**
-     * Clear the cache
+     * Clear the cache.
      *
-     * @returns {TileCache} this cache
+     * @returns {TileCache} Return `this`.
```
Comment (karimnaaji) [inline src/source/tile_cache.js:32]: For consistency with the rest of this change:
```suggestion
     * @returns {TileCache} Returns itself to allow for method chaining.
```

File: src/ui/map.js [hunk 6]
```diff
@@ -793,9 +791,8 @@ class Map extends Camera {
      * If the map's current pitch is lower than the new minimum,
      * the map will pitch to the new minimum.
      *
-     * @param {number | null | undefined} minPitch The minimum pitch to set (0-85).
-     *   If `null` or `undefined` is provided, the function removes the current minimum pitch (i.e. sets it to 0).
-     * @returns {Map} `this`
+     * @param {number | null | undefined} minPitch The minimum pitch to set (0-85). If `null` or `undefined` is provided, the function removes the current minimum pitch and resets it to 0.
```
Comment (karimnaaji) [inline src/ui/map.js:794]:
```suggestion
     * @param {number | null | undefined} `minPitch` The minimum pitch to set (0-85). If `null` or `undefined` is provided, the function removes the current minimum pitch and resets it to 0.
```

File: src/util/eased_variable.js [hunk 7]
```diff
@@ -37,7 +37,7 @@ class EasedVariable {
      *
      * @param timeStamp {number} Current time stamp.
      *
-     * @returns {boolean} true if ease in progress
+     * @returns {boolean} Returns true if ease in progress.
```
Comment (karimnaaji) [inline src/util/eased_variable.js:40]:
```suggestion
     * @returns {boolean} Returns `true` if ease in progress.
```

File: src/ui/map.js (hunk 6 above)
Comment (ryanhamley) [reply src/ui/map.js:794]: In other examples that name the parameter, it doesn't appear to be in back ticks

Comment (karimnaaji) [reply src/ui/map.js:794]: Yes you're right I read this too fast, it's the name of the parameter, all good here.

**Review by karimnaaji — APPROVED** [review]

**Pull request merged**